core/src/stored/btape.cc

```diff
@@ -205,99 +199,122 @@ int main(int margc, char* margv[])
           y32);
     exit(1);
   }
-  x64 = 123456789;
+
+  uint64_t x64 = 123456789;
   x64 = x64 << 32;
   x64 += 123456789;
   Bsnprintf(buf, sizeof(buf), "%llu", x64);
+
+  uint64_t y64;
   i = bsscanf(buf, "%llu", &y64);
   if (i != 1 || x64 != y64) {
     Pmsg3(-1, _("64 bit printf/scanf problem. i=%d x64=%llu y64=%llu\n"), i,
           x64, y64);
     exit(1);
   }
 
-  printf(_("Tape block granularity is %d bytes.\n"), TAPE_BSIZE);
-
   working_directory = "/tmp";
   MyNameIs(margc, margv, "btape");
   InitMsg(NULL, NULL);
 
   OSDependentInit();
 
-  while ((ch = getopt(margc, margv, "b:c:D:d:psv?")) != -1) {
-    switch (ch) {
-      case 'b': /* bootstrap file */
-        bsr = libbareos::parse_bsr(NULL, optarg);
-        //       DumpBsr(bsr, true);
-        break;
-
-      case 'c': /* specify config file */
-        if (configfile != NULL) { free(configfile); }
-        configfile = strdup(optarg);
-        break;
-
-      case 'D': /* specify director name */
-        if (DirectorName != NULL) { free(DirectorName); }
-        DirectorName = strdup(optarg);
-        break;
-
-      case 'd': /* set debug level */
-        if (*optarg == 't') {
-          dbg_timestamp = true;
-        } else {
-          debug_level = atoi(optarg);
-          if (debug_level <= 0) { debug_level = 1; }
-        }
-        break;
-
-      case 'p':
-        forge_on = true;
-        break;
+  CLI::App btape_app;
+  InitCLIApp(btape_app, "The Bareos Tape Manipulation tool.", 2000);
+
+  btape_app
+      .add_option(
+          "-b,--parse-bootstrap",
+          [](std::vector<std::string> vals) {
+            bsr = libbareos::parse_bsr(nullptr, vals.front().data());
+            return true;
+          },
+          "Specify a bootstrap file.")
+      ->check(CLI::ExistingPath)
+      ->type_name("<file>");
+
+  btape_app
+      .add_option(
+          "-c,--config",
+          [](std::vector<std::string> val) {
+            if (configfile != nullptr) { free(configfile); }
+            configfile = strdup(val.front().c_str());
+            return true;
+          },
+          "Use <path> as configuration file or directory.")
+      ->check(CLI::ExistingPath)
+      ->type_name("<path>");
+
+  std::string DirectorName;
+  btape_app
+      .add_option("-D,--director", DirectorName,
+                  "Specify a director name specified in the storage.\n"
+                  "Configuration file for the Key Encryption Key selection.")
+      ->type_name("<director>");
+
+  btape_app
+      .add_option("-d,--debug-level", debug_level,
+                  "Set debug level to <level>.")
+      ->check(CLI::NonNegativeNumber)
+      ->type_name("<level>");
+
+  btape_app.add_flag("--dt,--debug-timestamp", dbg_timestamp,
+                     "Print timestamp in debug output.");
+
+  btape_app.add_flag("-p,--proceed-io", forge_on,
+                     "Proceed inspite of IO errors");
+
+  btape_app.add_flag("-s,--turnoff-signals", turnoff_signals,
```

Comment:
```suggestion
  btape_app.add_flag("-s{false},--no-signals{false}", signals,
```
This should work also is is more intuitive (initializing signals=true, of course).

Reply:
You are right, it is possible and I tried it. Same goes for `!-s,!--no-signals`.
But if you use any of them, and you show the help, it looks like this:
```
-s{false},--no-signals{false}
```
which is in my opinion confusing